lib/hammer_cli/defaults.rb

```diff
@@ -0,0 +1,143 @@
+    end
+
+
+    def bad_input
+      print_message(_("You must specify value or a plugin name, cant specify both."))
+    end
+
+    def self.file_cant_be_created
+      print_message(_("Couldn't add file to ~/.hammer/cli.modules.d please create the path before defaults will be enabled."))
+    end
+
+    def list_all_defaults_message
+      unless HammerCLI::Settings.settings[:defaults].nil?
+        print_message("Hammer defaults list")
+        print_message("--------------------")
+        HammerCLI::Settings.settings[:defaults].each do |key,val|
+          if HammerCLI::Settings.settings[:defaults][key][:from_server]
+            print_message(key.to_s + " : " +  _("(provided by %{plugin})") % {:plugin => HammerCLI::Settings.settings[:defaults][key][:provider].to_s.split(':').first.gsub("HammerCLI", '')})
+          else
+            print_message(key.to_s + " : " + HammerCLI::Settings.settings[:defaults][key][:value].to_s)
+          end
+        end
+      else
+        print_message(_("No defaults file was found"))
+      end
+    end
+
+    autoload_subcommands
+  end
+
+
+  class Defaults
+    def self.register_provider(provider)
+      Defaults.providers[provider.keys[0]] = provider.values[0]
+    end
+
+    def self.providers
+      @@providers ||= {}
+    end
+
+    def self.add_defaults_to_conf(options, provider)
+      create_default_file unless !HammerCLI::Settings.settings[:defaults].nil?
+      path = "#{Dir.home}/.hammer/cli.modules.d/defaults.yml"
+      new_file = YAML.load_file(path)
+      options.each do |key, value|
+        key = key.to_sym
+        value.to_s.to_i if value.is_a? Integer
+        new_file[:defaults] = {} if new_file[:defaults].nil?
+        new_file[:defaults][key] = value ? {:value => value, :from_server => false} : {:from_server => true, :provider => provider}
+      end
+      File.open(path,'w') do |h|
+        h.write new_file.to_yaml
+      end
+    end
+
+    #this method will be overriden by plugins who wish to have a defaults params.
+    def self.get_defaults(option)
+      unless HammerCLI::Settings.settings[:defaults].nil? || HammerCLI::Settings.settings[:defaults][option.to_sym].nil?
+        if HammerCLI::Settings.settings[:defaults][option.to_sym][:from_server]
+          value =   Defaults.providers[HammerCLI::Settings.settings[:defaults][option.to_sym][:provider]].get_defaults(option.to_sym)
+          value
+        else
+          value = HammerCLI::Settings.settings[:defaults][option.to_sym][:value]
+          value
+        end
+      end
+    end
+
+    def self.create_default_file
+      path = "#{Dir.home}/.hammer/cli.modules.d/"
+      if Dir.exist?(path)
+        new_file = File.new(path + "/defaults.yml", "w")
+        new_file.write ":defaults:"
+        new_file.close
+      else
+        DefaultsCommand.file_cant_be_created
+      end
+
+    end
+
+  HammerCLI::MainCommand.subcommand "defaults", _("Defaults management"), HammerCLI::DefaultsCommand
+end
+end
```

Comment:
Wrong indentation. It would also be good to place the subcommand registration outside the `Defaults` class. I guess it was placed there by mistake as a result of the indentation issue.